lib/logger/utils.ts

```diff
@@ -46,7 +47,74 @@ const contentFields = [
   'yarnLockParsed',
 ];
 
+type ZodShortenedIssue =
+  | null
+  | string
+  | string[]
+  | {
+      [key: string]: ZodShortenedIssue;
+    };
+
+export function prepareZodIssues(input: unknown): ZodShortenedIssue {
+  // istanbul ignore if
+  if (!is.plainObject(input)) {
+    return null;
+  }
+
+  let err: null | string | string[] = null;
+  if (is.array(input._errors, is.string)) {
+    // istanbul ignore else
+    if (input._errors.length === 1) {
+      err = input._errors[0];
+    } else if (input._errors.length > 1) {
+      err = input._errors;
+    } else {
+      err = null;
+    }
+  }
+  delete input._errors;
+
+  if (is.emptyObject(input)) {
+    return err;
+  }
+
+  const output: Record<string, ZodShortenedIssue> = {};
+  const entries = Object.entries(input);
+  for (const [key, value] of entries.slice(0, 3)) {
+    const child = prepareZodIssues(value);
+    if (child !== null) {
+      output[key] = child;
+    }
+  }
+
+  if (entries.length > 3) {
+    output['___'] = `... ${entries.length - 3} more`;
+  }
+
+  return output;
+}
+
+export function prepareZodError(err: ZodError): Record<string, unknown> {
+  // istanbul ignore next
+  Object.defineProperty(err, 'message', {
+    get: () => 'Schema error',
+    set: (_) => {
+      _;
+    },
+  });
```

Comment:
why do we need this?  we return a new object anyways

Reply:
The `get message()` method of `ZodError` returns jsonified version of verbose inputs which makes stacktrace unreadable (as `err.trace` uses `err.message` internally)